core/src/main/java/org/elasticsearch/index/query/ScriptQueryBuilder.java

```diff
@@ -196,6 +196,11 @@ public float matchCost() {
                     };
                     return new ConstantScoreScorer(this, score(), twoPhase);
                 }
+
+                @Override
+                public boolean isCacheable(LeafReaderContext ctx) {
+                    return true;
```

Comment:
Should we return false and change the implementation of equals and hashCode if we still don't want to cache script query ?